x-pack/plugins/fleet/server/services/epm/registry/index.ts

```diff
@@ -256,32 +282,67 @@ function ensureContentType(archivePath: string) {
   return contentType;
 }
 
-export async function ensureCachedArchiveInfo(
-  name: string,
-  version: string,
-  installSource: InstallSource = 'registry'
-) {
-  const paths = getArchiveFilelist({ name, version });
-  if (!paths || paths.length === 0) {
-    if (installSource === 'registry') {
-      await getRegistryPackage(name, version);
-    } else {
-      throw new PackageCacheError(
-        `Package ${name}-${version} not cached. If it was uploaded, try uninstalling and reinstalling manually.`
-      );
+export async function fetchArchiveBuffer({
+  pkgName,
+  pkgVersion,
+  verify,
```

Comment:
`verify` should be renamed `shouldVerify` or something along those lines. At first glance I thought this was maybe a callback function but then realized it was a flag.